simtools/tests/test_visualize.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/python3
+
+import logging
+import os
+import sys
+import numpy as np
+from astropy import units as u
+from astropy.io import ascii
+from .. import visualize
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+def test_plot_1D():
+
+    logger.debug('Testing plot1D')
+
+    xTitle = 'Wavelength [nm]'
+    yTitle = 'Mirror reflectivity [%]'
+    headersType = {'names': (xTitle, yTitle),
+                   'formats': ('f8', 'f8')}
+    title = 'Test 1D plot'
+
+    dataIn = np.loadtxt('tests/testData/ref_200_1100_190211a.dat',
+                        usecols=(0, 1), dtype=headersType)
+
+    # Change y-axis to percent
+    if '%' in yTitle:
+        if np.max(dataIn[yTitle]) <= 1:
+            dataIn[yTitle] = 100*dataIn[yTitle]
+    data = dict()
+    data['Reflectivity'] = dataIn
+    for i in range(5):
+        newData = np.copy(dataIn)
+        newData[yTitle] = newData[yTitle]*(1 - 0.1*(i + 1))
+        data['{}%% reflectivity'.format(100*(1 - 0.1*(i + 1)))] = newData
+
+    plt = visualize.plot1D(data, title=title, palette='autumn')
+
+    plotFile = 'tests/testPlots/plot_1D.pdf'
+    if os.path.isfile(plotFile):
+        os.remove(plotFile)
+    plt.savefig(plotFile)
+    if not os.path.isfile(plotFile):
+        logger.critical('Did not create {}!'.format(plotFile))
+        sys.exit(1)
```

Comment:
Shouln't we raise and error here, instead of exit(1)?

Reply:
I didn't know which error to raise in this case, so I decided not to. However, I changed it now to raise a RuntimeError and removed the exit. Still not very informative, but better.

Reply:
We can create our own exceptions if we happen to need something specific.
In this case I'm not actually convinced that we need to do this verification at all.

Reply:
The verification is simply a way for pytest to know it failed. Usually you do an assert for it to fail, but I cannot assert anything for this test, so I do something like this. The RuntimeError is "general" enough for this purpose I guess.

Reply:
Ok, you are right.